site/en/posts/site-refactor-2022.md

```diff
@@ -0,0 +1,92 @@
+
+There was a third module we experimented with, [Vite Plugin Img](https://www.npmjs.com/package/vite-plugin-img), for doing the same kind of image manipulation we had previously done through Eleventy through Vite instead, but we wound up deciding against using it in production.
+
+## Managing Images
+
+Far and away the biggest contributor to our build times was how we managed our images. Images on our site have a number of requirements:
+
+- They need to be [lazy loaded](https://web.dev/learn/design/responsive-images/#loading-hints) so only images that are likely to be displayed are downloaded. Because of this, we also need to know their [size](https://web.dev/learn/design/responsive-images/#sizing-hints).
+- They need to be [responsive](https://web.dev/learn/design/responsive-images/#responsive-images-with-srcset), with the same image being scaled to multiple different sizes.
+- All of those images also need to be for multiple [image formats](https://web.dev/learn/design/picture-element/#image-formats), so we can have the best chance of serving the smallest size image for each user and device.
+- All generated images need to be optimized to reduce overall image size with minimal change in picture quality.
+- Any images whose source is an animated `.gif` needs to be converted to a silent, looping, autoplaying video, which are usually much smaller.
+
+These requirements exist so that we can, to the best of our ability, make sure our users are served only the media that's required for their browsing context, and only the most optimized versions of that media. What this meant in practice, though, was that for every individual image we had on our site, we needed to generate about 15 images in varying formats and potentially support GIF to video encoding. We also needed to rely on developers having a number of additional command-line packages installed that differed across operating systems.
+
+Our first attempt at this, [Eleventy Plugin Local Responsive Images](https://github.com/chromeos/static-site-scaffold-modules/tree/91285a3289358599df58b29190b80e8afd107d59/modules/eleventy-plugin-local-respimg) was able to do this pretty well; it kept an in-memory cache of images it had already optimized so it wouldn't have to do them twice in the same run, was super configurable, and didn't require special syntax to work; just an `img` tag that pointed to a local image, and it magically worked! For videos, too! It was built as an Eleventy transform, which meant that on each build of the site, it would need to do the transforms for all images across all rendered pages. This meant the first build of a watch task, for instance, was very slow, while subsequent builds were faster.
```

Comment:
```suggestion
Our first attempt at this, (the [Eleventy Plugin Local Responsive Images](https://github.com/chromeos/static-site-scaffold-modules/tree/91285a3289358599df58b29190b80e8afd107d59/modules/eleventy-plugin-local-respimg)) was able to do this pretty well; it kept an in-memory cache of images it had already optimized so it wouldn't have to do them twice in the same run, was super configurable, and didn't require special syntax to work; just an `img` tag that pointed to a local image, and it magically worked! For videos, too! It was built as an Eleventy transform, which meant that on each build of the site, it would need to do the transforms for all images across all rendered pages. This meant the first build of a watch task, for instance, was very slow, while subsequent builds were faster.
```

Reply:
I think I'd prefer to keep this as-is